cylc/flow/workflow_files.py

```diff
@@ -831,31 +832,36 @@ def check_deprecation(path, warn=True, force_compat_mode=False):
 
 
 def validate_workflow_name(
-    name: str, check_reserved_names: bool = False
+    name: str,
+    check_reserved_names: bool = False,
+    being_validated: Literal['workflow', 'workflow and run'] = 'workflow'
```

Comment:
I don't see the need for this argument; I would just keep it simple and drop any changes to this function

Reply:
I get your point, but if a user does

```
> cylc install --run-name  foo?
```

Then "invalid workflow name" seems unhelpful as a message. 

I'd be happy to modify the error messages without changing the functioning though.

Reply:
Nah, I agree, it's fine